web_monitoring/cli.py

```diff
@@ -359,6 +381,11 @@ def merge_worker_summaries(summaries):
 
         await uploader
 
+    # Our signal handler is specific to the context of this function, so
+    # reinstate the previous handlers before leaving.
+    signal.signal(signal.SIGINT, sigint_handler)
```

Comment:
Did you consider using a context manager or a `finally` block to ensure the original handlers are reinstated? Here's [an example of doing this with a context](https://github.com/NSLS-II/bluesky/blob/603ef4d8f713bd66ba3bf42d281f4c7903a6b7c5/bluesky/utils.py#L139-L175) but in this case `try/finally` is likely more expedient.

Reply:
Oh, good point! I should definitely at least do `finally`, but a context manager (or maybe a decorator for the whole function?) would wrap it up nicely.

```python
with SignalCancel(signal.SIGINT, signal.SIGTERM) as stop:
    ...
```